fusequery/query/src/datasources/local/csv_table.rs

```diff
@@ -59,33 +70,37 @@ impl ITable for CsvTable {
         Ok(self.schema.clone())
     }
 
-    fn read_plan(&self, _ctx: FuseQueryContextRef, _scan: &ScanPlan) -> Result<ReadDataSourcePlan> {
+    fn read_plan(&self, ctx: FuseQueryContextRef, _scan: &ScanPlan) -> Result<ReadDataSourcePlan> {
+        let output = Command::new("sh")
```

Comment:
This is bad.

1. `wc` and `cat` depend on target os env heavily.
2. Bad performance.


We can use `fs::metadata` instead.

Reply:
1. y
2. For the performance, `wc` almost the best(count words), see: https://benhoyt.com/writings/count-words/
Arrow csv reader has a [bounds](https://docs.rs/arrow/3.0.0/arrow/csv/reader/struct.Reader.html)(we can assign a [start, end] range line) , so I did the wc here.

The better solution is to partition by the chunk size(csv-size/cpus) and read the lines, a bit compex to handle the align the lines bounds but best performance.
Have found `fs:metadata`, but it may not be help here?

Reply:
1. y
2. For the performance, `wc` almost the best(count words), see: https://benhoyt.com/writings/count-words/
Arrow csv reader has a [bounds](https://docs.rs/arrow/3.0.0/arrow/csv/reader/struct.Reader.html)(we can assign a [start, end] range line) , so I did the wc here.

The better solution is to partition by the chunk size(csv-size/cpus) and read the lines, a bit compex to handle the align the lines bounds but best performance.
Have found `fs:metadata`, but it may not be help here?